src/live.rs

```diff
@@ -183,6 +215,81 @@ pub fn pxe_ignition_unwrap(config: PxeIgnitionUnwrapConfig) -> Result<()> {
     Ok(())
 }
 
+pub fn pxe_network_wrap(config: PxeNetworkWrapConfig) -> Result<()> {
+    if config.output.is_none() {
+        verify_stdout_not_tty()?;
+    }
+
+    let mut initrd = Initrd::default();
+    initrd_network_embed(&mut initrd, &config.keyfile)?;
+
+    write_live_pxe(&initrd, config.output.as_ref())
+}
+
+fn initrd_network_embed(initrd: &mut Initrd, keyfiles: &[String]) -> Result<()> {
+    for path in keyfiles {
+        let data = read(path).with_context(|| format!("reading {}", path))?;
+        let name = filename(path)?;
+        let path = format!("{}/{}", INITRD_NETWORK_DIR, name);
+        if initrd.get(&path).is_some() {
+            bail!("multiple input files named '{}'", name);
+        }
+        initrd.add(&path, data);
+    }
+    Ok(())
+}
+
+pub fn pxe_network_unwrap(config: PxeNetworkUnwrapConfig) -> Result<()> {
+    let stdin = io::stdin();
+    let f: Box<dyn Read> = if let Some(path) = &config.input {
+        Box::new(
+            OpenOptions::new()
+                .read(true)
+                .open(path)
+                .with_context(|| format!("opening {}", path))?,
+        )
+    } else {
+        Box::new(stdin.lock())
+    };
+    initrd_network_extract(
+        &Initrd::from_reader_filtered(f, &INITRD_NETWORK_GLOB)?,
+        config.directory.as_ref(),
+    )
+}
+
+fn initrd_network_extract(initrd: &Initrd, directory: Option<&String>) -> Result<()> {
+    let files = initrd.find(&INITRD_NETWORK_GLOB);
+    if files.is_empty() {
+        bail!("No embedded network settings.");
+    }
+    if let Some(dir) = directory {
+        create_dir_all(&dir)?;
+        for (path, contents) in files {
+            let path = Path::new(dir).join(filename(path)?);
+            OpenOptions::new()
+                .create_new(true)
+                .write(true)
+                .open(&path)
+                .with_context(|| format!("opening {}", path.display()))?
+                .write_all(contents)
+                .with_context(|| format!("writing {}", path.display()))?;
+            println!("{}", path.display());
+        }
+    } else {
+        for (i, (path, contents)) in files.iter().enumerate() {
+            if i > 0 {
+                println!();
+            }
+            println!("========== {} ==========", filename(path)?);
```

Comment:
We could make this a comment instead (`########## {} ##########`). That way, one can redirect the output to a file and get a syntactically correct NM keyfile in the case that there is only one.